### PR TITLE
[aspnetcore]Add initial support of Swashbuckle5 with asp.net core 3.0 rc1

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Changes
- Support SwashBuckle 5.0 with ASP.NET Core 3.0
- Endpoint support for ASP.NET Core 3.0
- Reverted setting of Json library because the generated code relies only on NewtonSoft Json library and is incompatible with MS Json Lib. NewtonSoft Json lib is default.
- Removed Endpoint routing switch, this is mandatory for routing.

@wing328 @mandrean @jimschubert

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/BasePathFilter.5_0.mustache [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        public string BasePath { get; }
+
+        /// <summary>
+        /// Apply the filter
+        /// </summary>
+        /// <param name="swaggerDoc">SwaggerDocument</param>
+        /// <param name="context">FilterContext</param>
+        public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
+        {
+            //swaggerDoc.BasePath = BasePath;
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/BasePathFilter.5_0.mustache]: Basepath setting is still not working. 
@wing328 @jimschubert 
any hints

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/GeneratePathParamsValidationFilter.5_0.mustache [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                    {
+                        swaggerParam.Required = true;
+                    }
+
+                    // Regex Pattern [RegularExpression]
+                    var regexAttr = attributes.FirstOrDefault(p => p.AttributeType == typeof(RegularExpressionAttribute));
+                    if (regexAttr != null)
+                    {
+                        //string regex = (string)regexAttr.ConstructorArguments[0].Value;
+                        //if (swaggerParam is NonBodyParameter)
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/GeneratePathParamsValidationFilter.5_0.mustache]: @wing328 @jimschubert 
Still need some work here, I am not sure how

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/GeneratePathParamsValidationFilter.5_0.mustache [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                        minLenght = (int)minLengthAttr.ConstructorArguments[0].Value;
+                    }
+
+                    var maxLengthAttr = attributes.FirstOrDefault(p => p.AttributeType == typeof(MaxLengthAttribute));
+                    if (maxLengthAttr != null)
+                    {
+                        maxLength = (int)maxLengthAttr.ConstructorArguments[0].Value;
+                    }
+
+                    //if (swaggerParam is NonBodyParameter)
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/GeneratePathParamsValidationFilter.5_0.mustache]: @wing328 @jimschubert 
Still need some work here, I am not sure how

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/GeneratePathParamsValidationFilter.5_0.mustache [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                    //}
+
+                    // Range [Range]
+                    var rangeAttr = attributes.FirstOrDefault(p => p.AttributeType == typeof(RangeAttribute));
+                    if (rangeAttr != null)
+                    {
+                        int rangeMin = (int)rangeAttr.ConstructorArguments[0].Value;
+                        int rangeMax = (int)rangeAttr.ConstructorArguments[1].Value;
+
+                        //if (swaggerParam is NonBodyParameter)
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Filters/GeneratePathParamsValidationFilter.5_0.mustache]: @wing328 @jimschubert 
Still need some work here, I am not sure how

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Project.csproj.mustache [hunk 5]
```diff
@@ -12,17 +12,7 @@
     <PackageId>{{packageName}}</PackageId>
   </PropertyGroup>
   <ItemGroup>
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Project.csproj.mustache]: Revert changes introduced for asp.net Core 3.0

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.3_0.mustache [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+                        Version = "{{#version}}{{{version}}}{{/version}}{{^version}}v1{{/version}}",
+                        Title = "{{#appName}}{{{appName}}}{{/appName}}{{^appName}}{{packageName}}{{/appName}}",
+                        Description = "{{#appName}}{{{appName}}}{{/appName}}{{^appName}}{{packageName}}{{/appName}} (ASP.NET Core 3.0)",
+                        Contact = new OpenApiContact()
+                        {
+                           Name = "{{#infoName}}{{{infoName}}}{{/infoName}}{{^infoName}}OpenAPI-Generator Contributors{{/infoName}}",
+                           Url = new Uri("{{#infoUrl}}{{{infoUrl}}}{{/infoUrl}}{{^infoUrl}}https://github.com/openapitools/openapi-generator{{/infoUrl}}"),
+                           Email = "{{#infoEmail}}{{{infoEmail}}}{{/infoEmail}}"
+                        },
+                        TermsOfService = "{{#termsOfService}}{{{termsOfService}}}{{/termsOfService}}"
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.3_0.mustache]: This is now a URI and we need to provide a link to the TOS. 
Needs commenting out for now

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.3_0.mustache [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+                    });
+                    c.CustomSchemaIds(type => type.FriendlyId(true));
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.3_0.mustache]: Friendly ID is no long available

File: modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.mustache [hunk 8]
```diff
@@ -60,16 +60,16 @@ namespace {{packageName}}
 
             // Add framework services.
             services
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.mustache]: Revert changes introduced for asp.net Core 3.0

**Comment by wing328** [comment]
cc  @A-Joshi for review as well as he's made a lot of contributions to the C# server generator.

**Pull request closed without merging**